src/main/java/io/strimzi/kafka/bridge/http/HttpSinkBridgeEndpoint.java

```diff
@@ -248,6 +248,16 @@ private void doDeleteConsumer(RoutingContext routingContext) {
     }
 
     private void doPoll(RoutingContext routingContext) {
+        if (!topicSubscriptionsPatternUsed && topicSubscriptions.isEmpty()) {
```

Comment:
not sure why we need this flag and it's not enough checking that the list is empty.
Subscribing or assigning is just filling some topics/partitions in the `topicSubscriptions` list.

Reply:
Look at https://github.com/strimzi/strimzi-kafka-bridge/blob/780dad5f5481b254a0b92ec6fe28e298be8ac30c/src/main/java/io/strimzi/kafka/bridge/http/HttpSinkBridgeEndpoint.java#L403

If we use `topics`, we do write topics into the `topicSubscriptions`. If we use `topic_pattern` we do not use `topicSubscriptions`.

Reply:
right! So maybe it could make sense to save the pattern instead of having a boolean. Maybe the pattern used by the HTTP client could return useful if in the future we want to provide a way for the user to retrieve it. Wdyt?